src/components/CreateMenu/BaseCreateMenu.js

```diff
@@ -26,54 +26,53 @@ const propTypes = {
     ).isRequired,
 
     ...windowDimensionsPropTypes,
+
+    // Trigger the item `onSelected` action immediately after press or after the modal hides
+    invokeActionImmediately: PropTypes.bool,
```

Comment:
I think the name of this isn't great... It's not clear to me when you would want the action to be triggered when clicking, or when the modal is closed. What would you think about using two separate props?

- `invokeActionWhenSelected`
- `invokeActionWhenModalClosed`

That would make it very explicit and I think that might clean up a little bit of the logic as well.

Reply:
Will update this tomorrow

Reply:
I've addressed this by adding an `executeActionMode` prop restricted to 2 values - on press or after modal hides
If you provide a wrong value by mistake you'll get a prop type warning (and an error)
IDE Intellisense would pickup and show supported options:
![image](https://user-images.githubusercontent.com/12156624/113847831-86f9a000-97a0-11eb-90f9-4470f9bd8864.png)

I didn't go with two separate props as you could set them both to `true` or both to `false` and it shouldn't be possible to configure to invoke `onSelect` twice - when pressed and after close, or never

Reply:
I'm leaving the comment unresolved so you can verify this works for you